src/Platforms/AbstractPlatform.php

```diff
@@ -427,6 +429,7 @@ protected function initializeCommentedDoctrineTypes()
         $this->doctrineTypeComments = [];
 
         foreach (Type::getTypesMap() as $typeName => $className) {
+            // and here too
```

Comment:
@morozov if I take a step backwards and try to understand what purpose `AbstractPlatform::doctrineTypeComments` and the attached method serve, I don't understand. To me, it looks like replacing all calls to `isCommentedDoctrineType($type)` with just `$doctrineType->requiresSQLCommentHint()` would do the trick, unless, and maybe this is where I'm missing something, somebody calls `markDoctrineTypeCommented`.

This logic was added very early in 28c7e09ddec068f0d3abc759858dcba8e05c1067 , and I see it wasn't like that from the start.

In https://github.com/doctrine/dbal/pull/2603, this was automated, and I think since then `markDoctrineTypeCommented` has become useless?

Searching for this org-wide only shows one call in a laminas module: https://github.com/search?q=org%3Adoctrine+markDoctrineTypeCommented&type=code

Maybe we should deprecate all this and fully rely on `requiresSQLCommentHint`?

Reply:
> I don't understand

I don't understand the purpose of this API either. It always look to me like a caching/memoization layer on top of `Type::requiresSQLCommentHint()` which looks redundant and causes issues, e.g. https://github.com/doctrine/dbal/issues/4983.

> Maybe we should deprecate all this and fully rely on `requiresSQLCommentHint`?

This looks right and aligns with https://github.com/doctrine/dbal/pull/5049#discussion_r759540026

Reply:
Yes. After reading the `git log`, it's clear that the API served a purpose, but no longer does since #2603. I'll work on that.